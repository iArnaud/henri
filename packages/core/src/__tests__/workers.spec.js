const BaseModule = require('../base/module');
const Henri = require('../henri');
const Workers = require('../5.workers');

describe('models', () => {
  beforeAll(async () => {
    this.henri = new Henri({
      runlevel: 5,
    });
    await this.henri.init();
  });

  afterAll(async () => {
    await this.henri.stop();
  });

  test('should be defined', () => {
    expect(this.henri.workers).toBeDefined();
  });

  test('should extend BaseModule', () => {
    expect(this.henri.workers).toBeInstanceOf(BaseModule);
  });

  test('should match snapshot', () => {
    const workers = new Workers();

    expect(workers).toMatchSnapshot();
  });

  test('should stop on reload', () => {
    const stop = jest.fn();

    this.henri.workers.workers['witness.js'] = {
      stop: stop,
    };

    this.henri.workers.reload();
    expect(stop).toHaveBeenCalledTimes(1);
  });
});
