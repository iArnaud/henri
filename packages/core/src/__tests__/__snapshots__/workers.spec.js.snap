// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`models should match snapshot 1`] = `
Workers {
  "consoleOnly": false,
  "files": Array [],
  "getFiles": [Function],
  "henri": undefined,
  "info": [Function],
  "init": [Function],
  "key": null,
  "name": "workers",
  "reload": [Function],
  "reloadable": true,
  "runlevel": 5,
  "setup": [Function],
  "start": [Function],
  "stop": [Function],
  "workers": Object {},
}
`;
